modules/msgsys/add.php

```diff
@@ -16,54 +16,51 @@
             $func->error(t('Du hast keinen Benutzer ausgew&auml;hlt'), "index.php?mod=msgsys&action=addbuddy");
         } else {
               $user[] = $_GET['userid'];
+
+            //init loop variables
+            $err = [];
+            $names1 = "";
+            $sux = [];
             foreach ($user as $buddyid) {
-                  // User already in list ?
-                  $existsinthelist = $db->qry("
-                    SELECT
-                      id
+                // User already in list ?
+                $userAlreadyAdded= $database->queryWithOnlyFirstRow(
+                    "SELECT id
                     FROM %prefix%buddys
-                    WHERE userid =%int%
-                    AND buddyid =%int%", $auth['userid'], $buddyid);
-
-                if ($db->num_rows() != "0") {
-                    $user_exist_in_the_list = 1;
-                }
+                    WHERE userid = ?
+                    AND buddyid = ?",
+                    [$auth['userid'], $buddyid]
+                );
 
-                  // Does the user exist ?
-                $exist = $db->qry("
-                  SELECT userid
+                // Does the user exist ?
+                $userExists = $database->queryWithOnlyFirstRow(
+                "SELECT userid
                   FROM %prefix%user
-                  WHERE userid =%int%", $buddyid);
-                if ($db->num_rows() != "0") {
-                    $user_exist = 1;
-                }
+                  WHERE userid = ?",
+                  [$buddyid]
+                  );
 
                 // Too many users in the list ?
-                $num = $db->qry("
-                  SELECT id
+                $user_num = $database->queryWithFullResult(
+                 "SELECT id
                   FROM %prefix%buddys
-                  WHERE userid =%int%", $auth['userid']);
-                $user_num = $db->num_rows();
-                if ($user_num >= 20) {
-                    $to_many_users = 1;
-                }
+                  WHERE userid = ?", [$auth['userid']]);
+
+                $too_many_users = sizeof($user_num) >= 20;
```

Comment:
```suggestion
                $too_many_users = count($user_num) >= 20;
```

`sizeof()` is only an alias from `count()`. See https://www.php.net/manual/de/function.sizeof.php
Lets use the original instead.